scimono-client/src/main/java/com/sap/scimono/client/ClientJacksonResolver.java

```diff
@@ -15,10 +15,20 @@
 @Consumes(APPLICATION_JSON_SCIM)
 @Produces(APPLICATION_JSON_SCIM)
 public class ClientJacksonResolver implements ContextResolver<ObjectMapper> {
+  private final boolean isUserNameOptional;
```

Comment:
As we talked this should be fixed. ClientJacksonResolver have to know nothing about user optional attributes. It can allow certain modification but not doing them by itself. Maybe, interceptor pattern can be used to execute entity specific configurations on object mapper. 
However, this may take time to be addressed, designed and implemented.
If this feature is urgent, merge it, but don't forget to address it afterwards. You can open an github issue as a reminder.